src/HDF5.jl

```diff
@@ -1378,12 +1378,11 @@ end
 
 # Helper for reading compound types
 function read_row(io::IO, membertype, membersize)
-    row = []
+    row = Vector{Any}()
```

Comment:
Also can use `Any[]`, I think tkelman mentioned that there have been some discussions of deprecated this type of constructor. May be best to use `Any[]` to be safe